packages/alfa-css/src/alphabet.ts

```diff
@@ -45,6 +46,12 @@ export type FunctionName = Readonly<{
 
 export type AtKeyword = Readonly<{ type: TokenType.AtKeyword; value: string }>;
 
+export type Hash = Readonly<{
+  type: TokenType.Hash;
+  typeFlag: "id" | "unrestricted";
```

Comment:
Given that these are the only options I'd do the following to make the type a little less awkward:

```ts
export type Hash = Readonly<{
  type: TokenType.Hash;
  unrestricted: boolean;
  value: string;
}>;
```

Reply:
Normally I would argue that we should follow the spec to the point so that if W3C introduces a third flag we can implement it easily. That being said, your solution is way more easy to read.